timemachine/lib/__init__.py

```diff
@@ -1,46 +1,62 @@
+from dataclasses import dataclass, field
+from typing import List
+
+import numpy as np
+from numpy.typing import NDArray
+
 from timemachine.lib import custom_ops
 
 # safe to pickle!
 
 
+@dataclass
 class LangevinIntegrator:
-    def __init__(self, temperature, dt, friction, masses, seed):
-
-        self.dt = dt
-        self.friction = friction
-        self.masses = masses
-        self.seed = seed
-        self.temperature = temperature
+    temperature: float
+    dt: float
+    friction: float
+    masses: NDArray[np.float64]
+    seed: int
 
     def impl(self):
         return custom_ops.LangevinIntegrator(self.masses, self.temperature, self.dt, self.friction, self.seed)
 
 
+@dataclass
 class VelocityVerletIntegrator:
-    def __init__(self, dt, masses):
-        self.dt = dt
+    dt: float
+    masses: NDArray[np.float64]
 
-        cb = dt / masses
+    cbs: NDArray[np.float64] = field(init=False)
+
+    def __post_init__(self):
+        cb = self.dt / self.masses
         cb *= -1
         self.cbs = cb
 
     def impl(self):
         return custom_ops.VelocityVerletIntegrator(self.dt, self.cbs)
 
 
+@dataclass
 class MonteCarloBarostat:
-
-    __slots__ = ("N", "temperature", "pressure", "group_idxs", "interval", "seed")
-
-    def __init__(self, N, pressure, temperature, group_idxs, interval, seed):
-        self.N = N
-        self.pressure = pressure
-        self.temperature = temperature
-        self.group_idxs = group_idxs
-        self.interval = interval
-        self.seed = seed
+    N: int
+    pressure: float
+    temperature: float
+    group_idxs: List[NDArray]
+    interval: int
+    seed: int
+    adaptive_scaling_enabled: bool = True
+    initial_volume_scale_factor: float = 0.0
 
     def impl(self, bound_potentials):
         return custom_ops.MonteCarloBarostat(
-            self.N, self.pressure, self.temperature, self.group_idxs, self.interval, bound_potentials, self.seed
+            self.N,
+            self.pressure,
+            self.temperature,
+            [g.tolist() for g in self.group_idxs],
```

Comment:
[nit, type surprise] -- is the `tolist()` conversion important? (was this previously brittle w.r.t. np.int32 vs. np.int64 type, and if so, would this be a useful change in other contexts, e.g. valence idxs?)

Reply:
The type of the `group_idxs` in the C++ API is `List[List[int]]`, although the wrapper seems to accept `NDArray` as well. Previously this was not causing any static type errors because the arguments of `timemachine.lib.MonteCarloBarostat` were not annotated, but this issue surfaced when I converted it to a dataclass.